doc/notebooks/temporal.ipynb

```diff
@@ -0,0 +1,276 @@
+    "\n",
+    "# Delete Zip\n",
+    "os.remove(\"../../data/NC_spm_temporal_workshop.zip\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Start GRASS GIS"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Ask GRASS GIS where its Python packages are.\n",
+    "sys.path.append(\n",
+    "    subprocess.check_output([\"grass\", \"--config\", \"python_path\"], text=True).strip()\n",
+    ")\n",
+    "\n",
+    "# Import GRASS packages\n",
+    "import grass.script as gs\n",
+    "import grass.jupyter as gj\n",
+    "\n",
+    "# Start GRASS Session\n",
+    "gj.init(\"../../data/grassdata\", \"NC_spm_temporal_workshop\", \"climate_2000_2012\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Set computational region to the elevation raster."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gs.run_command(\"g.region\", raster=\"elev_state_500m\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Create empty space time datasets"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gs.run_command('t.create', output='tempmean', type='strds',\n",
+    "                  temporaltype='absolute', title=\"Average temperature\",\n",
+    "                  description=\"Monthly temperature average in NC [deg C]\")\n",
+    "\n",
+    "gs.run_command('t.create', output='precip_sum', type='strds',\n",
+    "                  temporaltype='absolute', title=\"Preciptation\",\n",
+    "                  description=\"Monthly precipitation sums in NC [mm]\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Create list of rasters to be registered to empty space time datasets"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tempmean_list = gs.read_command(\"g.list\", type=\"raster\", pattern=\"*tempmean\", separator=\"comma\").strip()\n",
+    "precip_list = gs.read_command(\"g.list\", type=\"raster\", pattern=\"*precip\", separator=\"comma\").strip()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Register the rasters to the space time dataset created above"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gs.run_command(\"t.register\",\n",
+    "               input=\"tempmean\",\n",
+    "               type=\"raster\",\n",
+    "               start=\"2000-01-01\",\n",
+    "               increment=\"1 months\", \n",
+    "               maps=tempmean_list,\n",
+    "               flags=\"i\")\n",
```

Comment:
Since you are formatting these anyway like this, try the latest Black with extension for Jupyter and format the notebook too.

Reply:
Cool extension!